pyaedt/modules/report_templates.py

```diff
@@ -2444,40 +2431,569 @@ def _context(self):
+                "RJ",
                 False,
-                "0ps",
-                "WS",
+                str(self.rx_rj),
+                "SJ",
                 False,
-                self.time_start,
-            ],
-        ]
+                str(self.rx_sj),
+                "UNOI",
+                False,
+                str(self.rx_uniform_noise),
+                "USE_PRI_DIST",
+                False,
+                "0" if not self.enable_jitter_distribution else "1",
+            ]
+
+        arg = [
+            "NAME:Context",
+            "SimValueContext:=",
+            sim_context,
+        ]
+        return arg
+
+    @property
+    def _trace_info(self):
+        new_exprs = self.expressions if isinstance(self.expressions, list) else [self.expressions]
+        return ["X Component:=", "__UnitInterval", "Y Component:=", "__Amplitude", "Z Component:=", new_exprs]
+
+    @pyaedt_function_handler()
+    def create(self, plot_name=None):
+        """Create an eye diagram report.
+
+        Parameters
+        ----------
+        plot_name : str, optional
+            Plot name. The default is ``None``, in which case
+            the default name is used.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        if not plot_name:
+            self.plot_name = generate_unique_name("Plot")
+        else:
+            self.plot_name = plot_name
+        self._post.oreportsetup.CreateReport(
+            self.plot_name,
+            self.report_category,
+            self.report_type,
+            self.setup,
+            self._context,
+            self._convert_dict_to_report_sel(self.variations),
+            self._trace_info,
+        )
+        self._post.plots.append(self)
+        self._is_created = True
+
+        return True
+
+    @pyaedt_function_handler()
+    def eye_mask(
+        self,
+        points,
+        xunits="ns",
+        yunits="mV",
+        enable_limits=False,
+        upper_limit=500,
+        lower_limit=-500,
+        color=(0, 255, 0),
+        xoffset="0ns",
+        yoffset="0V",
+        transparency=0.3,
+    ):
+        """Create an eye diagram in the plot.
+
+        Parameters
+        ----------
+        points : list
+            Points of the eye mask in the format ``[[x1,y1,],[x2,y2],...]``.
+        xunits : str, optional
+            X points units. The default is ``"ns"``.
+        yunits : str, optional
+            Y points units. The default is ``"mV"``.
+        enable_limits : bool, optional
+            Whether to enable the upper and lower limits. The default is ``False``.
+        upper_limits float, optional
+            Upper limit if limits are enabled. The default is ``500``.
+        lower_limits str, optional
+            Lower limit if limits are enabled. The default is ``-500``.
+        color : tuple, optional
+            Mask in (R, G, B) color. The default is ``(0, 255, 0)``.
```

Comment:
Do you perhaps want to use the same format earlier where RGB is used to be consistent?